consensus-types/hdiff/state_diff.go

```diff
@@ -0,0 +1,2101 @@
+	justificationBits           byte                                                        // override.
+	slot                        primitives.Slot                                             // override.
+	fork                        *ethpb.Fork                                                 // override.
+	latestBlockHeader           *ethpb.BeaconBlockHeader                                    // override.
+	blockRoots                  [fieldparams.BlockRootsLength][fieldparams.RootLength]byte  // zero or override.
+	stateRoots                  [fieldparams.StateRootsLength][fieldparams.RootLength]byte  // zero or override.
+	historicalRoots             [][fieldparams.RootLength]byte                              // append only.
+	eth1Data                    *ethpb.Eth1Data                                             // override.
+	eth1DataVotes               []*ethpb.Eth1Data                                           // append only or override.
+	eth1DepositIndex            uint64                                                      // override.
+	randaoMixes                 [fieldparams.RandaoMixesLength][fieldparams.RootLength]byte // zero or override.
+	slashings                   [fieldparams.SlashingsLength]int64                          // algebraic diff.
+	previousEpochAttestations   []*ethpb.PendingAttestation                                 // override.
+	currentEpochAttestations    []*ethpb.PendingAttestation                                 // override.
+	previousJustifiedCheckpoint *ethpb.Checkpoint                                           // override.
+	currentJustifiedCheckpoint  *ethpb.Checkpoint                                           // override.
+	finalizedCheckpoint         *ethpb.Checkpoint                                           // override.
+	// Altair Fields
+	previousEpochParticipation []byte               // override.
+	currentEpochParticipation  []byte               // override.
+	inactivityScores           []uint64             // override.
+	currentSyncCommittee       *ethpb.SyncCommittee // override.
+	nextSyncCommittee          *ethpb.SyncCommittee // override.
+	// Bellatrix
+	executionPayloadHeader interfaces.ExecutionData // override.
+	// Capella
+	nextWithdrawalIndex          uint64
+	nextWithdrawalValidatorIndex primitives.ValidatorIndex
+	historicalSummaries          []*ethpb.HistoricalSummary // append only.
+	// Electra
+	depositRequestsStartIndex     uint64
+	depositBalanceToConsume       primitives.Gwei
+	exitBalanceToConsume          primitives.Gwei
+	earliestExitEpoch             primitives.Epoch
+	consolidationBalanceToConsume primitives.Gwei
+	earliestConsolidationEpoch    primitives.Epoch
+
+	pendingDepositIndex            uint64
+	pendingPartialWithdrawalsIndex uint64
+	pendingConsolidationsIndex     uint64
+	pendingDepositDiff             []*ethpb.PendingDeposit
+	pendingPartialWithdrawalsDiff  []*ethpb.PendingPartialWithdrawal
+	pendingConsolidationsDiffs     []*ethpb.PendingConsolidation
```

Comment:
maybe it would be nicer to order them as this, it got me thinking the order matters here (but it doesnt) 

```suggestion
	pendingDepositIndex            uint64
	pendingDepositDiff             []*ethpb.PendingDeposit	
	pendingPartialWithdrawalsIndex uint64
	pendingPartialWithdrawalsDiff  []*ethpb.PendingPartialWithdrawal	
	pendingConsolidationsIndex     uint64
	pendingConsolidationsDiffs     []*ethpb.PendingConsolidation
```